### PR TITLE
remove RSN4c, add RTS4b (channels#release)

### PR DESCRIPTION
**Pull request by tiholic**
`channels#release` in should raise error in `realtime` for specific states and should return without error in `rest`

[RSN4c](https://docs.ably.io/client-lib-development-guide/features/#RSN4c) should be put under [RTS4](https://docs.ably.io/client-lib-development-guide/features/#RTS4)

Discussion: https://ably-real-time.slack.com/archives/C030C5YLY/p1615715227000800

### CODE REVIEW

File: content/client-lib-development-guide/features.textile
Comment (SimonWoolf) [inline content/client-lib-development-guide/features.textile:494]: So now these two spec items contradict each other - if you call release on an attached channel, should it detach then release, or raise an error?

My suggestion in the slack thread was just get rid of RSN4c entirely

**Review by SimonWoolf — CHANGES_REQUESTED** [review]

**Comment by tbedford** [comment]
@tiholic @SimonWoolf @QuintinWillison - is there any follow up on this? Are we able to resolve? If not should I close this PR? Thanks!

File: content/client-lib-development-guide/features.textile
Comment (tiholic) [reply content/client-lib-development-guide/features.textile:494]: @SimonWoolf I lost context after a long gap. Removing `RTS4b` is what we are looking for, is it?

**Comment by tiholic** [comment]
@tbedford we can land the PR after make necessary changes. Thanks for your patience :)

File: content/client-lib-development-guide/features.textile
Comment (SimonWoolf) [reply content/client-lib-development-guide/features.textile:494]: yes

**Comment by QuintinWillison** [comment]
This pull request will need more work. See [this internal discussion](https://ably.atlassian.net/wiki/spaces/SDK/pages/683409414/Client+Library+Features+Spec?focusedCommentId=1544126559#comment-1544126559) in relation to the safe removal of spec points, where those spec points are at the end of a list.

File: content/client-lib-development-guide/features.textile
Comment (tiholic) [reply content/client-lib-development-guide/features.textile]: @SimonWoolf removed

**Comment by tiholic** [comment]
@QuintinWillison @tbedford @SimonWoolf necessary changes have been made.

**Review by tbedford — APPROVED** [review]

**Review by QuintinWillison — APPROVED** [review]

File: content/client-lib-development-guide/features.textile
Comment (QuintinWillison) [reply content/client-lib-development-guide/features.textile]: Resolving this conversation on @SimonWoolf's behalf.

**Comment by QuintinWillison** [comment]
@tbedford This is ready to land but I'll leave that to your team, when you're ready.

**Review by tbedford — APPROVED** [review]

**Comment by tbedford** [comment]
@QuintinWillison - Looks good to me, however I can't merge until the feedback from @SimonWoolf is attended to. Thanks.

**Review by SimonWoolf — APPROVED** [review]

**Pull request merged**
